### PR TITLE
docs: add an FAQ section

### PR DESCRIPTION
**Pull request by grouville**
Adds an FAQ section answering the most common mistakes / questions encountered when doing support: emptying buildkit cache, having better logs, always executing an action, chaining actions together and default values in Cue

Pair: @slumbering 
Signed-off-by: guillaume

### CODE REVIEW

**Comment by marcosnils** [comment]
Gave it initial pass. LGTM! Would be great if @jpadams can also :eyes:

**Review by marcosnils — APPROVED** [review]

**Review by shykes — CHANGES_REQUESTED** [review]
Great content, but please split it up into individual articles. A single FAQ page is not sustainable and does not leverage our knowledge base-style article system.

You can start by moving them under "guides". Then if that grows too long, you can create a "FAQ" category.

**Comment by grouville** [comment]
> long

Do you mean separating each content in an article inside the `guides`, but still keep that FAQ page ? Or also removing my FAQ page, and just moving each article inside the `guide` section ?

**Comment by grouville** [comment]
@shykes, rebased, and changed according to your requests

File: docs/guides/1230-chain-actions.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    input: _dockerCLI.output
+    always: true
+    script: contents: #"""
+        echo "bonjour"
+        """#
+    env: HACK: "\(test.success)" // <== HACK: CHAINING of action happening here
+}
+```
+
+`foo` and `bar` are similar actions. I don't want to rely on the `input-output` model but still want to force a dependency between them. The easiest way is to create an environment variable that relies on the other action's success
```
Comment (TomChv) [inline docs/guides/1230-chain-actions.md:54]: I'm not sure that using `I` is a good idea in documentation.
Maybe you wanted to say `If you do not`?

File: docs/guides/1230-chain-actions.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    always: true
+    script: contents: #"""
+        echo "bonjour"
+        """#
+}
+
+// Main action
+bar: bash.#Run & {
+    input: _dockerCLI.output
+    always: true
```
Comment (TomChv) [inline docs/guides/1230-chain-actions.md:46]: Why `always`, it is not related to the page?

File: docs/guides/1230-chain-actions.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+On above example, `_dockerCLI` gets executed first, as the corresponding DAG shows that it is required for the action `foo` to be processed.
+
+This is the `input-output` model: `foo`'s input is `_dockerCLI`'s output.
+
+We currently don't support explicit dependencies at the moment (one of our top priority). But, if you need one, here is how your can hack your way around it:
+
+```cue
+foo: bash.#Run & {
+    input: _dockerCLI.output
+    always: true
```
Comment (TomChv) [inline docs/guides/1230-chain-actions.md:37]: Why `always`, it is not related to the page?

File: docs/guides/1230-chain-actions.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    script: contents: #"""
+        echo "bonjour"
```
Comment (TomChv) [inline docs/guides/1230-chain-actions.md:39]: Why not `hello`?

File: docs/guides/1230-chain-actions.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        """#
+}
+
+// Main action
+bar: bash.#Run & {
+    input: _dockerCLI.output
+    always: true
+    script: contents: #"""
+        echo "bonjour"
```
Comment (TomChv) [inline docs/guides/1230-chain-actions.md:48]: Why not `hello`?

File: docs/guides/1230-chain-actions.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        bash: {}
+    }
+}
+
+// Main action
+foo: bash.#Run & {
+    input: _dockerCLI.output // <== CHAINING of action happening here
+    always: true
+    script: contents: #"""
+        echo "bonjour"
```
Comment (TomChv) [inline docs/guides/1230-chain-actions.md:23]: Why not `hello`?

File: docs/guides/1230-chain-actions.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+_dockerCLI: alpine.#Build & {
+    packages: {
+        bash: {}
+    }
+}
+
+// Main action
+foo: bash.#Run & {
+    input: _dockerCLI.output // <== CHAINING of action happening here
+    always: true
```
Comment (TomChv) [inline docs/guides/1230-chain-actions.md:21]: Why `always`?

File: docs/guides/1230-chain-actions.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+# Chaining actions
+
+Dependencies are materialized at runtime, when your Cue files are parsed and the corresponding DAG gets generated:
+
+```cue
+// Prerequisite action that runs when `test` is being called
+_dockerCLI: alpine.#Build & {
+    packages: {
+        bash: {}
+    }
```
Comment (TomChv) [inline docs/guides/1230-chain-actions.md:15]: Could be directly `packages: bash: _`
And I think using `{}` isn't the standard, use `_` instead

File: docs/guides/1231-default-values-cue.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
+---
+slug: /1231/default-values-cue
+displayed_sidebar: europa
+---
+
+# Default values and optional fields in Cue
```
Comment (TomChv) [inline docs/guides/1231-default-values-cue.md:6]: It's two topics, and I think it's already in the [CUE doc](https://docs.dagger.io/1215/what-is-cue/#default-values-and-the-nature-of-inheritance).
Why duplication?

**Review by TomChv — COMMENTED** [review]
That's a great documentation!

I left few comments to improve it.

File: docs/guides/1227-empty-buildkit-cache.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+There are two ways of emptying the BuildKit cache:
+
+- Run your action with the `--no-cache` option:
+
+```console
+dagger do <your-action> --no-cache
+```
+
+- Stop and remove the buildkitd container and remove its associated volume:
```
[the remark this answers is not in the document]
Comment (slumbering) [reply docs/guides/1227-empty-buildkit-cache.md:16]: Or you could also say: Stop and remove the buildkitd container **then** remove its associated volume:

**Review by shykes — APPROVED** [review]
Thanks Guillaume! This looks good to me, feel free to incorporate the outstanding suggestions as a follow-up.

**Comment by shykes** [comment]
NOTE: this will need rebase before merging, there are conflicts (I'm guessing on article IDs)

File: docs/guides/1227-empty-buildkit-cache.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
+---
+slug: /1227/empty-buildkit-cache
+displayed_sidebar: europa
```
Comment (gerhard) [inline docs/guides/1227-empty-buildkit-cache.md:3]: This should be `displayed_sidebar: '0.2'`

File: docs/guides/1228-better-logs.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
+---
+slug: /1228/better-logs
+displayed_sidebar: europa
```
Comment (gerhard) [inline docs/guides/1228-better-logs.md:3]: This should be `displayed_sidebar: '0.2'`

File: docs/guides/1228-better-logs.md [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+
+# Log formats
+
+Dagger exposes 2 logging format options:
+
+- `--log-format auto|plain|tty|json`
+
+The default mode is `auto`. If you want to keep each actions' logs, use the `plain` mode
+
+- `--log-level panic|fatal|error|warn|info|debug|trace`
```
Comment (gerhard) [inline docs/guides/1228-better-logs.md:14]:
```diff
- --log-level panic|fatal|error|warn|info|debug|trace
+ --log-level <panic|fatal|error|warn|info|debug|trace>
```

We can do this post-merge, not a blocker

File: docs/guides/1229-always-execute.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
+---
+slug: /1229/always-execute
+displayed_sidebar: europa
```
Comment (gerhard) [inline docs/guides/1229-always-execute.md:3]: This should be `displayed_sidebar: '0.2'`

File: docs/guides/1229-always-execute.md [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+# Always executing an action
+
+Dagger implemented a way to tell Buildkit not to rely on cache for a specific action.
+
+The `docker.#Run` and `core.#Exec` definitions have an `always` field:
+
+```cue
+// If set to true, the cache will never be triggered for that specific action.
+always: bool | *false
```
Comment (gerhard) [inline docs/guides/1229-always-execute.md:14]: Is the convention, `type | *defaultValue` or `*defaultValue | type`?

I have seen us use both and I would like us to be consistent.

File: docs/guides/1230-chain-actions.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
+---
+slug: /1230/chain-actions
+displayed_sidebar: europa
```
Comment (gerhard) [inline docs/guides/1230-chain-actions.md:3]: This should be `displayed_sidebar: '0.2'`

File: docs/guides/1231-default-values-cue.md (hunk 9 above)
Comment (gerhard) [reply docs/guides/1231-default-values-cue.md:6]: Smaller doc pages sound like a good idea to me. If this page is cohesive, it's OK to have duplication.

File: docs/guides/1231-default-values-cue.md [hunk 17]
```diff
@@ -0,0 +1,54 @@
+---
+slug: /1231/default-values-cue
+displayed_sidebar: europa
```
Comment (gerhard) [inline docs/guides/1231-default-values-cue.md:3]: This should be `displayed_sidebar: '0.2'`

**Comment by gerhard** [comment]
This requires a few changes, but we can blast through them in less than 30 mins.
If you want, I am open to doing it together so that we can get this in.
Great stuff @grouville!

File: docs/guides/1227-empty-buildkit-cache.md (hunk 11 above)
Comment (grouville) [reply docs/guides/1227-empty-buildkit-cache.md:3]: `'displayed_sidebar: 0.2' returns me an error

**Comment by grouville** [comment]
@gerhard, with @slumbering we managed to create the FAQ category 😇

**Comment by gerhard** [comment]
Taking another look in the next 30 mins.

**Pull request merged**